apps/dev/src/commands/feeds/read/adfs.ts

```diff
@@ -9,9 +9,43 @@ import {
 import { renderTui, drawTable } from '@blocksense/base-utils/tty';
 import { listEvmNetworks, readEvmDeployment } from '@blocksense/config-types';
 import { AggregatedDataFeedStoreConsumer } from '@blocksense/contracts/viem';
+import { formatTimestamp } from '../../utils';
 
 const availableNetworks = await listEvmNetworks();
 
+export const parseADFSData = (hexData: string) => {
+  const cleanHex = hexData.startsWith('0x') ? hexData.slice(2) : hexData;
+  const valuePart = cleanHex.slice(0, 48);
+  const valueHex = '0x' + valuePart.padEnd(64, '0');
+  let value: bigint = 0n;
+  try {
+    value = BigInt(valueHex);
+  } catch {
+    // ignore
+  }
+  let timestamp: bigint | null = null;
+  let timestampHex: string | null = null;
+  if (cleanHex.length >= 64) {
+    timestampHex = '0x' + cleanHex.slice(48, 64);
+    try {
+      timestamp = BigInt(timestampHex);
+    } catch {
+      timestamp = null;
+    }
+  }
+  const formattedTimestamp =
+    timestamp !== null ? formatTimestamp(timestamp) : null;
```

Comment:
Similarly, for the timestamp, we know it is in the data and it is the rest of the bytes, so we do not need complicated checks and parsing. Just:

```suggestion
  const timestampHex = '0x' + cleanHex.slice(48);
  const timestamp = BigInt(timestampHex);
  const formattedTimestamp = formatTimestamp(timestamp);
```